### PR TITLE
fix(graphql): include sort key(s) in many to many directive relation model

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Currently @manyToMany directive is not including sort key(s) in the relation table and erroring out on gql compile, this is stopping customers from using manyToMany directive when the table contains a sort key. This change will fix that issue.

#### Issue #, if available

Fixes https://github.com/aws-amplify/amplify-cli/issues/9526

#### Description of how you validated changes
- yarn test
- manual test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-relational-transformer/src/schema.ts [hunk 1]
```diff
@@ -430,3 +430,24 @@ export function getPartitionKeyField(ctx: TransformerContextProvider, object: Ob
...
+  const fieldMap = new Map<string, FieldDefinitionNode>();
+
+  for (const field of outputObject.fields!) {
+    fieldMap.set(field.name.value, field);
+  }
+
+  for (const field of outputObject.fields!) {
+    for (const directive of field.directives!) {
+      if (directive.name.value === 'primaryKey') {
+        let values: ListValueNode = directive.arguments?.find(arg => arg.name.value === 'sortKeyFields')?.value as ListValueNode;
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply packages/amplify-graphql-relational-transformer/src/schema.ts:446]: can `values.values` be undefined?

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts [hunk 2]
```diff
@@ -174,6 +174,73 @@ test('valid schema', () => {
...
+      id: ID!
+      models: [ModelA] @manyToMany(relationName: "ModelAModelB")
+    }`;
+  const transformer = createTransformer();
+  const out = transformer.transform(inputSchema);
+  expect(out).toBeDefined();
+  const schema = parse(out.schema);
+  validateModelSchema(schema);
+
+  expect(out.schema).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts:195]: Added steps to validate relation model.

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts [hunk 3]
```diff
@@ -174,6 +174,73 @@ test('valid schema', () => {
...
+      sortId: ID!
+      models: [ModelA] @manyToMany(relationName: "ModelAModelB")
+    }`;
+  const transformer = createTransformer();
+  const out = transformer.transform(inputSchema);
+  expect(out).toBeDefined();
+  const schema = parse(out.schema);
+  validateModelSchema(schema);
+
+  expect(out.schema).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts:218]: Added steps to validate relation model.

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts [hunk 4]
```diff
@@ -174,6 +174,73 @@ test('valid schema', () => {
...
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-many-to-many-transformer.test.ts:243]: Added steps to validate relation model.

File: packages/amplify-graphql-relational-transformer/src/schema.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-graphql-relational-transformer/src/schema.ts]: Updated to const. `values.values` can't be undefined.

**Review by lazpavel — APPROVED** [review]
LGTM

**Comment by Dennis-Dekker** [comment]
Hi, do you have an indication of when this fix will be merged and released? Thanks!

**Review by lazpavel — APPROVED** [review]

**Comment by ivadenis** [comment]
Can you please merge this!

**Review by marcvberg — APPROVED** [review]
LGTM, corresponding codegen change also in flight

**Pull request merged**